WIP: Each position has portfolios

marcelometal opened the pull request.

marcelometal committed:
Each position has portfolios

kov commented:
I think this makes sense! I would prefer if we also make the Position calculation use the (symbol, portfolio) combo as ID along with this commit, what do you think?

kov commented:
mongodb magic to convert existing collection after this change:

```javascript
db.getCollection('operations').aggregate([
    {
        "$project": {
            "portfolios": [ "$portfolio" ],
            "assetType": 1,
            "symbol":  1,
            "price":  1,
            "quantity": 1,
            "fees": 1,
            "time": 1,
            "type": 1,
            "broker": 1,
        }
    },
    {
        "$out": "operations_new"
    }
])
```

kov committed:
Add CRUD for Portfolio

kov committed:
Fix referencing of portfolios on operations

To properly implement that react-admin expects the provider to return only
the specified ids when calling the "get all" API, so implement that on the
backend as well.

kov commented:
> ```js
>             "portfolios": [ "$portfolio" ],
> ```

Actually scratch that heh. It looks like react-admin really wants to talk in terms of ids, so we need to start using the ids anyway. With the changes I just pushed I'm able to create, edit and list operations with several portfolios quite nicely. Now just need to change the Position code to generate positions related to a (symbol, portfolio) combo.

marcelometal committed:
Litle cosmetic change
